public/js/category_list_script.js

```diff
@@ -0,0 +1,35 @@
+"use strict";
+
+$(".category-edit").click(function () {
+    var cat_id = $(this).parent().parent().data('category-id');
+    var initial_name = $(this).parent().parent().children('.category-name').text();
+    // console.log(initial_name);
+    var element = $('<input/>')
+        .attr('type', 'text')
+        .addClass('name_input')
+        .val(initial_name)
+        .wrap($('<td/>'));
+    // var html_string = '<td> <input type="text" class="name_input" value='+initial_name+' autofocus></td>';
+    $(this).parent().parent().children('.category-name').html('').append(element);
+    $(this).parent().parent().children('.category-name').children('.name_input').focus()
+    $('.name_input').change(function () {
+        var new_name = $(this).val();
+        $(this).parent().replaceWith('<td class="category-name">'+new_name+'</td>');
+        $.ajax({
+            url : '/category_edit',
+            type: "post",
+            data: JSON.stringify({id: cat_id, name: new_name}),
+            contentType: "application/json"
+        });
+    });
+    $('.name_input').blur(function () {
+        var new_name = $(this).val();
+        $(this).parent().replaceWith('<td class="category-name">'+new_name+'</td>');
+        $.ajax({
+            url : '/category_edit',
+            type: "post",
+            data: JSON.stringify({id: cat_id, name: new_name}),
+            contentType: "application/json"
```

Comment:
@daria-sieroshtan 
This is copy-paste
Moreover, if you have "blur" handler, you don't need "change" handler at all